lib/rubocop/result_cache.rb

```diff
@@ -188,14 +194,24 @@ def rubocop_checksum
             .select { |path| File.file?(path) }
             .sort!
             .each do |path|
-              content = File.binread(path)
-              digest << Zlib.crc32(content).to_s # mtime not reliable
+              digest << digest(path)
             end
           digest << RuboCop::Version::STRING << RuboCop::AST::Version::STRING
           digest.hexdigest
         end
     end
 
+    def digest(path)
+      if path.end_with?(*DL_EXTENSIONS)
+        # shared libraries often contain timestamps of when
+        # they were compiled and other non-stable data
+        Zlib.crc32(File.basename(path)).to_s
+      else
+        content = File.binread(path)
+        Zlib.crc32(content).to_s # mtime not reliable
```

Comment:
There's a little bit of duplication here. You could move `content =` to before the `if` and then have `Zlib.crc32(content).to_s` as the last line of the method, common for both branches.